llvm_objects.py

```diff
@@ -171,68 +175,114 @@ def __eq__(self, other):
+                            fl_var = Float(llwrap.LLVMConstRealGetDouble(arg, signed_ptr))
                             w_module.constants.set_variable(addr, fl_var)
         return args
+
+class W_BrInstruction(W_BaseInstruction):
+    ''' Represents a wrapper for an LLVMValueRef that is a br instruction. '''
+
+    def __init__(self, l_instr, w_module):
+        W_BaseInstruction.__init__(self, l_instr, w_module)
+        self.l_bb_uncond = llwrap.LLVMValueAsBasicBlock(self.l_operands[0])
+
+class W_ConditionalInstruction(W_BaseInstruction):
+    ''' Represents a wrapper for an LLVMValueRef that is a conditional
+        br instruction. '''
+
+    def __init__(self, l_instr, w_module):
+        W_BaseInstruction.__init__(self, l_instr, w_module)
+        self.condition = llwrap.LLVMGetCondition(l_instr)
+        self.l_bb_true = llwrap.LLVMValueAsBasicBlock(self.l_operands[2])
+        self.l_bb_false = llwrap.LLVMValueAsBasicBlock(self.l_operands[1])
+
+class W_ICmpInstruction(W_BaseInstruction):
+    ''' Represents a wrapper for an LLVMValueRef that is an icmp
+        instruction. '''
+
+    def __init__(self, l_instr, w_module):
+        W_BaseInstruction.__init__(self, l_instr, w_module)
+        self.icmp_predicate = llwrap.LLVMGetICmpPredicate(l_instr)
+
+class W_PhiInstruction(W_BaseInstruction):
+    ''' Represents a wrapper for an LLVMValueRef that is a phi
+        instruction. '''
+
+    def __init__(self, l_instr, w_module):
+        W_BaseInstruction.__init__(self, l_instr, w_module)
+        self.count_incoming = llwrap.LLVMCountIncoming(l_instr)
+        self.incoming_block = []
+        self.incoming_value = []
+        for i in range(self.count_incoming):
+            self.incoming_block.append(llwrap.LLVMGetIncomingBlock(l_instr, i))
+            self.incoming_value.append(llwrap.LLVMGetIncomingValue(l_instr, i))
+
+class W_CallInstruction(W_BaseInstruction):
+    ''' Represents a wrapper for an LLVMValueRef that is a call
+        instruction. '''
+
+    def __init__(self, l_instr, w_module):
+        W_BaseInstruction.__init__(self, l_instr, w_module)
+        if not llwrap.LLVMIsDeclaration(self.l_operands[-1]):
+            self.func_param_count = llwrap.LLVMCountParams(self.l_operands[-1])
+            self.l_func_params = []
+            for index in range(self.func_param_count):
+                self.l_func_params.append(llwrap.LLVMGetParam(self.l_operands[-1], index))
+        else:
+            # assume the instruction is either printf or puts (this is checked by the
+            # the interpreter)
+            self.l_string_format_ref = llwrap.LLVMGetOperand(self.l_operands[0], 0)
+            self.func_name = rffi.charp2str(llwrap.LLVMGetValueName(self.l_operands[-1]))
+
+def _get_instruction(opcode, l_instruction, w_module):
+    ''' Returns the appropriate wrapper for the specified instruction. '''
+
+    if opcode == llwrap.LLVMBr:
```

Comment:
You typically use `is` to check type via identity, but if these are ints you are comparing, then weird stuff will happen.